src/blob/core.rs

```diff
@@ -321,23 +347,24 @@ where
         &mut self,
         key: &K,
         only_if_presented: bool,
-    ) -> Result<bool> {
+    ) -> Result<DeleteResult> {
         if !only_if_presented || self.index.get_any(key).await?.is_found() {
-            self.push_deletion_record(key).await?;
-            Ok(true)
+            self.push_deletion_record(key).await
         } else {
-            Ok(false)
+            Ok(DeleteResult { dirty_bytes: self.file.dirty_bytes(), success: false })
         }
     }
 
-    async fn push_deletion_record(&mut self, key: &K) -> Result<()> {
+    async fn push_deletion_record(&mut self, key: &K) -> Result<DeleteResult> {
         let on_disk = self.index.on_disk();
         if on_disk {
             self.load_index().await?;
         }
         let record = Record::deleted(key)?;
         let header = self.write_mut(key, record).await?;
-        self.index.push_deletion(key, header)
+        self.index.push_deletion(key, header)?;
```

Comment:
Just noticed a bug. `write_mut` already adds the header to `index`. This addition is the second one. So this results in 2 identical headers inside the index